### PR TITLE
basis for passport auth and socket communication

### PR DESCRIPTION
**Pull request by Gattermeier**
lots of code to review. there were quite a bit of merge conflicts

### CODE REVIEW

File: client/app/app.js [hunk 1]
```diff
@@ -1,65 +1,73 @@
...
-      }else{ //anything other than 
-        //func: check if player is defined first
-        if($window.player!==undefined){
-          $window.player.pauseVideo();
-        } //if
-      } //if(url)
-    } //if
+    }); //on(routeChangeStart)
+  })
+  .run(function($rootScope, $location) {
```
Comment (whamsicore) [inline client/app/app.js:67]: Why a second .run?

File: server.js [hunk 2]
```diff
@@ -25,17 +28,37 @@ mongoose.connect(config.database, function(err) {
   }
   app.use("/", express.static(path.join(__dirname, 'client')));
 
+  app.use(require('body-parser').urlencoded({
```
Comment (whamsicore) [inline server.js:31]: What is this for?

File: client/app/app.js (hunk 1 above)
Comment (Gattermeier) [inline client/app/app.js:67]: @whamsicore accidental, pls ignore, I'll remove it in the next pull request.

File: server.js (hunk 2 above)
Comment (Gattermeier) [inline server.js:31]: @whamsicore the body parser is a part of connect middleware, it allows to create new middleware for express and attach content to req.body

**Pull request merged**